x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/520_connector_secret_delete.yml

```diff
@@ -0,0 +1,71 @@
+setup:
+  - skip:
+      version: " - 8.12.99"
+      reason: Introduced in 8.13.0
+
+---
+'Delete connector secret - admin':
+  - do:
+      connector_secret.post:
+        body:
+          value: my-secret
+  - set: { id: id }
+  - match: { id: $id }
+
+  - do:
+      connector_secret.delete:
+        id: $id
+  - match: { deleted: true }
+
+  - do:
+      connector_secret.get:
+        id: $id
+      catch: missing
+
+---
+'Delete connector secret - user with privileges':
+  - skip:
+      features: headers
+
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      connector_secret.post:
+        body:
+          value: my-secret
+  - set: { id: id }
+  - match: { id: $id }
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      connector_secret.delete:
+        id: $id
+  - match: { deleted: true }
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      connector_secret.get:
+        id: $id
+      catch: missing
+
+---
+'Delete connector secret - user without privileges':
+  - skip:
+      features: headers
+
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      connector_secret.post:
+        body:
+          value: my-secret
+  - set: { id: id }
+  - match: { id: $id }
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVucHJpdmlsZWdlZDplbnRzZWFyY2gtdW5wcml2aWxlZ2VkLXVzZXI=" }  # unprivileged
+      connector_secret.get:
+        id: $id
+      catch: unauthorized
+
+---
+'Get connector secret - Missing secret id':
```

Comment:
can we also test that null/ "" empty id causes 400 validation error?

Reply:
If this value is null or empty the request can't actually go through, so apparently the null check is redundant. The only GET or DELETE requests that are supported are `/_connector/_secret/{id}`. Making id null or empty changes the request to `/_connector/_secret/` which throws a 405.

There's no inbuilt method for this (e.g. `catch: method_not_allowed` isn't supported in the codebase). We also don't check this for similar endpoints like delete connector (we do a [null check](https://github.com/elastic/elasticsearch/blob/666774a8659643c6e779647023f26a5f4e54cce8/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/DeleteConnectorAction.java#L55-L57) but we don't have a test for it in its [yaml test](resources/rest-api-spec/test/entsearch/320_connector_delete.yml), probably for the same reason I'm having trouble adding one).

Do you think we should just remove the redundant null check? Or just continue forward without a yaml test for this?

Reply:
In that case I think it's fine to go ahead without the yaml test 👍